org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java

```diff
@@ -380,7 +380,7 @@ public void firePropertyChangeEvent(String name, Object oldValue,
 		// important: create intermediate array to protect against listeners
 		// being added/removed during the notification
 		final Object[] list = getListeners();
-		if (list.length == 0) {
+		if (list.length == 0 || Objects.equal(oldValue, newValue)) {
```

Comment:
this differs from other pref stores impls. eg.
ScopedPreferenceStore where this seems to be c&ped from

Reply:
Also my impression. If in doubt, better change the listener impl that does too many things when the event is indicating a no-op.

Please note that platform classes have logic like this:
```
if (changed > 0)
			masterPreferenceStore.firePropertyChangeEvent(getPreferencePrefix(), null, null);
```
This event would be swallowed with the proposed change.

Reply:
Thanks for the pointers. I'll have a second look.